library/src/main/java/com/bumptech/glide/load/resource/gif/GifDrawable.java

```diff
@@ -309,7 +309,23 @@ public void setLoopCount(int loopCount) {
         }
 
         if (loopCount == LOOP_INTRINSIC) {
-            maxLoopCount = decoder.getLoopCount();
+            int netscapeIterationCount = decoder.getNetscapeIterationCount();
```

Comment:
@TWiStErRob @sjudd
Although I was not sure the difference between the responsibilities that `GifDecoder` and `GifDrawable` have, I have sorted out my thoughts as follows:

- `GifDecoder` reflects the _raw_ data structure of the GIF file.
- `GifDrawable` interprets the data structure exposed by `GifDecoder` so that Glide can handle it easily.

If the above thoughts are different from yours, I can move the code snippet to `GifDecoder`.
In that case, I'm thinking to implement it with the following manner. Any comments are welcome!

- Leave `getNetscapeIterationCount()` which gets _raw_ netscape iteration count
  so that other interpretations of the Netscape iteration count (like Firefox) can be used.
- Create a following method in `GifDecoder`.  
  (Is `Resulting` an appropriate naming?)
    
    ```java
    public int getResultingLoopCount() {
        if (header.loopCount == GifHeader.NETSCAPE_ITERATION_COUNT_DOES_NOT_EXIST) {
            return 1;
        }
        return header.loopCount + 1;
    }
    ```
- In this place, the code snippet becomes as follows.
  Note that a conditional branch can't be eliminated completely because the value of `FOREVER` is different between `GifDecoder` and `GifDrawable`. 
    
    ```java
    ...
    int resultingLoopCount = decoder.getResultingLoopCount();
    if (resultingLoopCount == GifHeader.NETSCAPE_ITERATION_COUNT_FOREVER) {
        maxLoopCount = LOOP_FOREVER;
    } else {
        maxLoopCount = resultingLoopCount;
    }
    ...
    ```